homeassistant/components/remote_calendar/coordinator.py

```diff
@@ -0,0 +1,72 @@
+"""Data UpdateCoordinator for the Remote Calendar integration."""
+
+from datetime import timedelta
+import logging
+
+from httpx import ConnectError, HTTPStatusError, UnsupportedProtocol
+from ical.calendar import Calendar
+from ical.calendar_stream import IcsCalendarStream
+from ical.exceptions import CalendarParseError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+type RemoteCalendarConfigEntry = ConfigEntry[RemoteCalendarDataUpdateCoordinator]
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(days=1)
+
+
+class RemoteCalendarDataUpdateCoordinator(DataUpdateCoordinator[Calendar]):
+    """Class to manage fetching calendar data."""
+
+    config_entry: RemoteCalendarConfigEntry
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: RemoteCalendarConfigEntry,
+    ) -> None:
+        """Initialize data updater."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+            always_update=True,
+        )
+        self._etag = None
+        self._client = get_async_client(hass)
+        self._url = config_entry.data[CONF_URL]
+
+    async def _async_update_data(self) -> Calendar:
+        """Update data from the url."""
+        try:
+            res = await self._client.get(self._url, follow_redirects=True)
+        except (UnsupportedProtocol, ConnectError, HTTPStatusError, ValueError) as err:
+            raise UpdateFailed(
+                translation_domain=DOMAIN,
+                translation_key="unable_to_fetch",
+                translation_placeholders={"err": str(err)},
+            ) from err
+        else:
```

Comment:
The else nesting is not needed since all exception blocks are returning/raising. Remove the else and de-dent the code blocks to save horizontal whitespace and have less nesting here and elsewhere. 

(The reason it is there in the config flow is because the except cases re-render the form)

Reply:
https://github.com/home-assistant/core/pull/138862/commits/4969c0cf72f7a9a9bab2c1a6876327716898c448